spec/services/streaming_service_provider_data_service_spec.rb

```diff
@@ -0,0 +1,315 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe StreamingServiceProviderDataService do
+  describe 'get_providers' do
+    let(:movie_title) { 'Fake' }
+    let(:movie) { build(:movie, title: movie_title, release_date: '2023-01-01') }
+    let(:movie_args) do
+      {
+        tmdb_id: movie.tmdb_id,
+        title: movie.title,
+        media_type: 'movie',
+        media_format: 'movie',
+        release_date: movie.release_date
+      }
+    end
+
+    describe 'when the API does not return any data' do
+      it 'returns our default providers with a pay_model of "try"' do
+        allow(Tmdb::Client).to receive(:request).and_return(nil)
+
+        results = StreamingServiceProviderDataService.get_providers(movie_args)
+        aggregate_failures "all default providers appear with 'try'" do
+          expect(results.find {|r| r[:name] == 'Netflix'}[:pay_model] ).to eq('try')
+          expect(results.find {|r| r[:name] == 'Amazon Prime Video'}[:pay_model] ).to eq('try')
+          expect(results.find {|r| r[:name] == 'Amazon Video'}[:pay_model] ).to eq('try')
+          expect(results.find {|r| r[:name] == 'YouTube'}[:pay_model] ).to eq('try')
+          expect(results.find {|r| r[:name] == 'Vudu'}[:pay_model] ).to eq('try')
+        end
+      end
+    end
+
+    describe 'when the API does return results' do
+      describe 'but the results do not have free, flatrate, rent, or buy pay model options' do
+        it 'returns our default providers with a pay_model of "try"' do
+          response_data = {
+            results: {
+              US: {
+                ads:
+                  [{:logo_path=>"/5OAb2w.jpg", :provider_id=>634, :provider_name=>"Netflix", :display_priority=>26},
+                  {:logo_path=>"/eWp5Ld.jpg", :provider_id=>43, :provider_name=>"Amazon Video", :display_priority=>38},
+                  {:logo_path=>"/xL9SUR.jpg", :provider_id=>358, :provider_name=>"YouTube", :display_priority=>51}],
+                subscribe:
+                  [{:logo_path=>"/peURlL.jpg", :provider_id=>2, :provider_name=>"Vudu", :display_priority=>4},
+                  {:logo_path=>"/peURlL.jpg", :provider_id=>2, :provider_name=>"Amazon Prime Video", :display_priority=>4}],
```

Comment:
i'm sure you got this from copying the actual api response, but we should add rubocop at some point soon so that it would automatically fix these hashrockets.

i wonder if there some example starter rubocop setups out there we could grab so it wouldn't be a huge undertaking?

Reply:
I know there is a way to only run on modified files ([here is one post about it](https://dev.to/risafj/running-rubocop-only-on-modified-files-in-a-project-with-no-rubocop-2ioe)). I'm wondering if there is a way to configure rubocop itself to run on modified only. 🤔  That's something to look into for another PR.

Reply:
Neat. Okay, I created an issue to investigate some sort of rubocop solution: https://github.com/mikevallano/tmdb-moviequeue/issues/358